example/onnx/test_super_resolution.py

```diff
@@ -0,0 +1,84 @@
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""Testing super_resolution model conversion"""
+from __future__ import absolute_import as _abs
+from __future__ import print_function
+from collections import namedtuple
+import logging
+import numpy as np
+from PIL import Image
+import mxnet as mx
+from mxnet.test_utils import download
+import mxnet.contrib.onnx as onnx_mxnet
+
+# set up logger
+logging.basicConfig()
+LOGGER = logging.getLogger()
+LOGGER.setLevel(logging.INFO)
+
+def download_onnx_model():
+    """Download the onnx model"""
+    model_url = 'https://s3.amazonaws.com/onnx-mxnet/examples/super_resolution.onnx'
+    download(model_url, 'super_resolution.onnx')
+
+def import_onnx():
+    """Import the onnx model into mxnet"""
+    LOGGER.info("Converting onnx format to mxnet's symbol and params...")
+    sym, params = onnx_mxnet.import_model('super_resolution.onnx')
+    assert sym is not None
+    assert params is not None
+    return sym, params
+
+def get_test_image():
+    """Download and process the test image"""
+    # Load test image
+    input_image_dim = 224
+    img_url = 'https://s3.amazonaws.com/onnx-mxnet/examples/super_res_input.jpg'
+    download(img_url, 'super_res_input.jpg')
+    img = Image.open('super_res_input.jpg').resize((input_image_dim, input_image_dim))
+    img_ycbcr = img.convert("YCbCr")
+    img_y, img_cb, img_cr = img_ycbcr.split()
+    input_image = np.array(img_y)[np.newaxis, np.newaxis, :, :]
+    return input_image, img_cb, img_cr
+
+def perform_inference((sym, params), (input_img, img_cb, img_cr)):
+    """Perform inference on image using mxnet"""
+    # create module
+    mod = mx.mod.Module(symbol=sym, data_names=['input_0'], label_names=None)
+    mod.bind(for_training=False, data_shapes=[('input_0', input_img.shape)])
+    mod.set_params(arg_params=params, aux_params=None)
+
+    # run inference
+    batch = namedtuple('Batch', ['data'])
+    mod.forward(batch([mx.nd.array(input_img)]))
+
+    # Save the result
+    img_out_y = Image.fromarray(np.uint8(mod.get_outputs()[0][0][0].
+                                         asnumpy().clip(0, 255)), mode='L')
+
+    result_img = Image.merge(
+        "YCbCr", [img_out_y,
+                  img_cb.resize(img_out_y.size, Image.BICUBIC),
+                  img_cr.resize(img_out_y.size, Image.BICUBIC)]).convert("RGB")
+    output_img_dim = 672
+    assert result_img.size == (output_img_dim, output_img_dim)
```

Comment:
@eric-haibin-lin Does this assert match? I don't think it assesses the task properly.

Reply:
@marcoabreu added more asserts in the import_onnx() method where the actual model import is happening.

Reply:
Thanks. But please also verify the output

Reply:
the model is supposed to convert a low resolution image into a high resolution image with specific dimensions, so I have asserted for the size of the result image. What else will I have to assert for?

Reply:
I would (don't know if it's feasible) compare the similarity between a simply upscaled picture and the produced picture. If it differs too much, this could indicate that something went wrong and we didn't actually create an upscale picture but just scrambled bits in picture format. 

But when I'm thinking about it... This is an example that is supposed to be used by users as a reference, right? I think we should rather write a test on top of the examples instead of overloading it with verification logic as it could cause issues differentiating the necessary pieces from verification logic.

What do you think?